### PR TITLE
fix: dashboard insight updating

### LINKED ISSUE #10263: Dashboard insights don't update after changing insight definition
I defined a funnel with the wrong event, so I clicked edit and updated the funnel step. Then I returned to the dashboard where I was viewing it, but it had the old funnel steps. Refreshing the page fixed it, but it should ideally update without me having to refresh.

Repro:

1. View a dashboard
2. Click Edit from the insight on the dashboard
3. Change a funnel step
4. Return to view dashboard and see how the updated insight doesn't update in the dashboard until a browser refresh

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

There were no tests on whether changing an insight would update dashboards it was on. 

Changing an insight did not update dashboards :/

## Changes

* removes the code that made too many assumptions and was called incorrectly
* adds a test that if `dashboardsModel.actions.updateDashboardInsight` is called the updates make it to the dashboard
* renames `dashboardsModel.actions.updateDashboardItem` to `dashboardsModel.actions.updateDashboardInsight`

### move between dashboards updating dashboards

move-between-dashboards

### add and remove dashboards modal

fixes #10263 

## How did you test this code?

added developer tests and running locally

### CODE REVIEW

File: frontend/src/scenes/insights/insightLogic.test.ts [hunk 1]
```diff
@@ -712,10 +727,66 @@ describe('insightLogic', () => {
...
+    test.skip('updateInsight can add and remove a dashboard', async () => {
+        savedInsightsLogic.mount()
+        logic = insightLogic({
+            dashboardItemId: Insight43,
+        })
+        logic.mount()
+
+        logic.actions.updateInsight({ ...partialInsight43, dashboards: [8, 9] })
+        await expectLogic(logic).toDispatchActions(['updateInsightSuccess']).toFinishAllListeners()
+        // const expectedResponseWithAddedDashboards = patchResponseFor(
```
Comment (pauldambra) [inline frontend/src/scenes/insights/insightLogic.test.ts:767]: @mariusandra this is the test where I believe I'm passing in the right action args but it isn't seeing that the action was dispatched

Was getting late so hopefully I'm just being silly

File: frontend/src/scenes/dashboard/dashboardLogic.ts [hunk 2]
```diff
@@ -189,27 +188,37 @@ export const dashboardLogic = kea<dashboardLogicType>({
...
                     }
 
                     if (state) {
                         const itemIndex = state.items.findIndex((i) => i.short_id === item.short_id)
                         const newItems = state.items.slice(0)
+
                         if (itemIndex >= 0) {
-                            newItems[itemIndex] = mergeWithDashboardTile(item, newItems[itemIndex])
+                            if (item.dashboards?.includes(props.id)) {
+                                newItems[itemIndex] = { ...newItems[itemIndex], ...item }
```
Comment (pauldambra) [inline frontend/src/scenes/dashboard/dashboardLogic.ts:208]: Ultimately this is the fix

`mergeWithDashboardTile` didn't work :'/

File: frontend/src/scenes/insights/insightLogic.test.ts (hunk 1 above)
Comment (pauldambra) [reply frontend/src/scenes/insights/insightLogic.test.ts]: It's better tested than it was and this is getting in the way of a fix :/

**Review by mariusandra — APPROVED** [review]
Looks like code 😅

**Pull request merged**